amoro-common/src/main/java/org/apache/amoro/process/ManagedProcess.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.amoro.process;
+
+/**
+ * ManagedProcess is the interface for AMS modules to manage process.
+ *
+ * @param <T>
+ */
+public interface ManagedProcess<T extends ProcessState> extends AmoroProcess<T> {
+
+  /** Submit this process to ResourceContainer. */
+  void submit();
+
+  /** Mark this process as completed, trigger callbacks */
+  void complete();
+
+  /** Mark this process as failed, trigger callbacks */
+  void complete(String failedReason);
```

Comment:
```suggestion
  void failed(String failedReason);
```

Reply:
This is quite different.
The complete(String failReason) is called by task completion phase. and failing could happen at any phase(like submitting), so I think complete(String failReason) is more reasonable here which means only completion callback would use this method

Reply:
how about define one method with a flag indicate the process is success or not

Reply:
What's the point?

Reply:
Because these two methods seem to perform different execution logic, but the two method names are the same, so I recommend differentiating the names

Reply:
I mean what is the difference from 'define one method with a flag indicate the process is success or not' 
Could you give an example